KSystemInformer/util.c

```diff
@@ -1393,3 +1393,38 @@ NTSTATUS KphGuardGrantSuppressedCallAccess(
                                          &targetListInfo,
                                          sizeof(targetListInfo));
 }
+
+/**
+ * \brief UNICODE_STRING version of wcsrchr
+ * 
+ * Note: the output string will use the input strings Buffer!
+ *
+ * \param[in] String to look within, might be NULL.
+ * \param[in] ToFind charakter to find.
+ * \param[out] Out an alocated empty unicode string to set values of, must not be NULL.
+ *
+ * \return TRUE on success
+ */
+_IRQL_requires_max_(PASSIVE_LEVEL)
+BOOLEAN KphUSrchr(
+    _In_ PUNICODE_STRING String,
+    _In_ WCHAR ToFind,
+    _Out_ PUNICODE_STRING Out
+    )
+{
+    if (!String || !String->Buffer)
+        return FALSE;
+
+    WCHAR* ptr = &String->Buffer[String->Length / sizeof(WCHAR)];
+    while (ptr > String->Buffer) {
+        if (*--ptr == ToFind)
+            break;
+    }
+
+    USHORT len = (USHORT)((ptr - String->Buffer) * sizeof(WCHAR));
+    Out->Buffer = ptr;
+    Out->Length = String->Length - len;
+    Out->MaximumLength = String->MaximumLength - len;
+
+    return len > 0;
```

Comment:
There are style issues here, please declare variables at the top of scope, and do not use K&R brace style.